### PR TITLE
Fix sea.js issues #962, #957, #956

### PR DESCRIPTION
**Pull request by mimiza**
Fixes these issues:
(https://github.com/amark/gun/issues/956)
(https://github.com/amark/gun/issues/957)
(https://github.com/amark/gun/issues/962)
BEFORE:
- user.recall() uses alias/pass -> not reliable, only works with auth(alias,pass), doesn't work with auth(pair)
- user.auth(pair, cb) -> cb is not a function

AFTER:
- user.recall() uses pair -> more reliable, works with auth(pair)
- user.auth(pair, cb) -> works

### CODE REVIEW

**Comment by capitalistdog** [comment]
This appears to have resolved the initial issue in our tests, but we're investigating issues with overwriting existing user data (unsure if related still)

**Comment by amark** [comment]
Yaay!

@mimiza @capitalistdog @jabis can we do a call today/tomorrow or Wed+ to review & merge? I think jabis saw a couple issues that might break some apps? So hopefully on call we can work that out & move to publish.

**Comment by mimiza** [comment]
We all have different timezones. But yeah, please arrange a call, but please let me know what time. Or we can just discuss here, it's ok for me. My English listening skill is not so fluent, just warning.

**Comment by mimiza** [comment]
I have just switched to ancient JS technique, so it would be more compatible with those stone-age browsers. Removed (...args), replaced with 'arguments' :D

**Comment by amark** [comment]
@mimiza yay!

Timezones I've found early CA often works for everybody?

@mimiza @capitalistdog @jabis Wednesday 10am?

**Comment by mimiza** [comment]
By CA do you mean California (GMT-7) timezone? Mine is GMT+7 which is 12 hours foward. So your 10AM is my 10PM. Ok Wed 10PM.

**Comment by mimiza** [comment]
@amark Oh but where do we meet? Meething I guess? Somewhere else?

File: sea.js [hunk 1]
```diff
@@ -825,16 +825,21 @@
       return gun;
     }
     // now that we have created a user, we want to authenticate them!
-    User.prototype.auth = function(alias, pass, cb, opt){
+    User.prototype.auth = function(){
+      const alias = typeof arguments[0] === 'string' ? arguments[0] : null
```
Comment (sirpy) [inline sea.js:829]: i'd keep arguments as part of the function signature, otherwise its not very readable or dev friendly

**Comment by amark** [comment]
@jabis @mimiza @sirpy @capitalistdog + whoever else:

I'll be on:

https://us.meething.space/?room=gun

in about 15min from now. Is it still a good time?

**Comment by amark** [comment]
we switched to this: https://whereby.com/danlugo92

**Comment by amark** [comment]
w0000000h00000000 🍾 🎈 🎉 🎈 🎉 🎈 🎉 🎈 🎉  you are an AWESOME ROLE MODEL for OPEN SOURCE :D :D :D

**Comment by mimiza** [comment]
Thank you all ❤️😊

File: sea.js (hunk 1 above)
Comment (amark) [reply sea.js:829]: yes, let's refactor it.

**Pull request merged**